tests/test_wheel.py

```diff
@@ -45,3 +52,62 @@ def test_find_metadata_files():
     ]
     candidates = wheel.Wheel.find_candidate_metadata_files(names)
     assert expected == candidates
+
+
+def test_read_valid(example_wheel):
+    """Test reading a valid wheel file"""
+    metadata = example_wheel.read().decode().splitlines()
+    assert "Name: twine" in metadata
+    assert "Version: 1.5.0" in metadata
+
+
+def test_read_non_existent_wheel_file_name():
+    """Test reading a wheel file which doesn't exist"""
+
+    file_name = "/foo/bar/baz.whl"
+    with pytest.raises(exceptions.InvalidDistribution) as excinfo:
+        wheel.Wheel(file_name)
+
+    assert "No such file: {}".format(file_name) == str(excinfo.value)
+
+
+def test_read_invalid_wheel_extension():
+    """Test reading a wheel file without a .whl extension"""
+
+    file_name = os.path.join(os.path.dirname(__file__), "fixtures/twine-1.5.0.tar.gz")
+    with pytest.raises(exceptions.InvalidDistribution) as excinfo:
+        wheel.Wheel(file_name)
+
+    assert "Not a known archive format: {}".format(file_name) == str(excinfo.value)
+
+
+def test_read_wheel_without_metadata(capsys, tmpdir):
+    """Test reading a wheel file with empty metadata"""
+
+    def create_empty_metadata_wheel(name, version):
+        def add_file(path, text):
+            contents = text.encode("utf-8")
+            z.writestr(path, contents)
+            records.append((path, text, str(len(text))))
+
+        dist_info = "{}-{}.dist-info".format(name, version)
```

Comment:
Also, since we're on Python 3, all use of `.format` could be f-strings:

```suggestion
        dist_info = f"{name}-{version}.dist-info"
```

Reply:
Agreed, I can change that, but like`PathLib`, even `f-strings` were introduced in Python3.6 . Will this cause a problem for folks using older Python3 versions?